From 6272396a05148abf60b305466ae6408c484db7a3 Mon Sep 17 00:00:00 2001
From: Lukasz Majczak <lukaszm@marvell.com>
Date: Tue, 11 Feb 2020 13:35:37 +0100
Subject: [PATCH 1095/1239] octeontx: octeontx2: Print preserved memory size

Print separate log for secure and non-secure memory region.

Change-Id: Id6951b827e4e173cd37e1f839600c564fe72a3fd
Signed-off-by: Lukasz Majczak <lukaszm@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/23207
Tested-by: Chandrakala Chavva <cchavva@marvell.com>
Reviewed-by: Chandrakala Chavva <cchavva@marvell.com>
---
 board/Marvell/octeontx/board-fdt.c  | 23 ++++++++++++++++-------
 board/Marvell/octeontx2/board-fdt.c | 23 ++++++++++++++++-------
 2 files changed, 32 insertions(+), 14 deletions(-)

diff --git a/board/Marvell/octeontx/board-fdt.c b/board/Marvell/octeontx/board-fdt.c
index 2344b3258d..7ee64076e7 100644
--- a/board/Marvell/octeontx/board-fdt.c
+++ b/board/Marvell/octeontx/board-fdt.c
@@ -187,6 +187,7 @@ u64 fdt_get_preserved_mem_size(unsigned int node)
 	const char *str = NULL;
 	char prop_name[34] = {};
 	u64 preserved_mem_size = 0;
+	u64 total_preserved_mem_size = 0;
 
 	bdk_node = fdt_get_bdk_node();
 	if (!bdk_node)
@@ -194,17 +195,25 @@ u64 fdt_get_preserved_mem_size(unsigned int node)
 
 	snprintf(prop_name, 34, "DDR-CONFIG-PRESERVE-NON-SECURE.N%d", node);
 	str = fdt_getprop(fdt, bdk_node, prop_name, &len);
-	if (str)
-		preserved_mem_size += (strtoul(str, NULL, 16) << 20);
+	if (str) {
+		preserved_mem_size = (strtoul(str, NULL, 16) << 20);
+		total_preserved_mem_size += preserved_mem_size;
+		if (preserved_mem_size)
+			printf("Preserved non-secure memory: 0x%llx bytes\n",
+			       preserved_mem_size);
+	}
 
 	snprintf(prop_name, 34, "DDR-CONFIG-PRESERVE-SECURE.N%d", node);
 	str = fdt_getprop(fdt, bdk_node, prop_name, &len);
-	if (str)
-		preserved_mem_size += (strtoul(str, NULL, 16) << 20);
+	if (str) {
+		preserved_mem_size = (strtoul(str, NULL, 16) << 20);
+		total_preserved_mem_size += preserved_mem_size;
+		if (preserved_mem_size)
+			printf("Preserved secure memory: 0x%llx bytes\n",
+			       preserved_mem_size);
+	}
 
-	if (preserved_mem_size)
-		printf("Total memory preserved region: 0x%llx bytes\n", preserved_mem_size);
-	return preserved_mem_size;
+	return total_preserved_mem_size;
 }
 
 const char *fdt_get_board_serial(void)
diff --git a/board/Marvell/octeontx2/board-fdt.c b/board/Marvell/octeontx2/board-fdt.c
index ff1cb4af6b..7d161ce8f7 100644
--- a/board/Marvell/octeontx2/board-fdt.c
+++ b/board/Marvell/octeontx2/board-fdt.c
@@ -100,6 +100,7 @@ u64 fdt_get_preserved_mem_size(unsigned int node)
 	const char *str = NULL;
 	char prop_name[34] = {};
 	u64 preserved_mem_size = 0;
+	u64 total_preserved_mem_size = 0;
 
 	bdk_node = fdt_get_bdk_node();
 	if (!bdk_node)
@@ -107,17 +108,25 @@ u64 fdt_get_preserved_mem_size(unsigned int node)
 
 	snprintf(prop_name, 34, "DDR-CONFIG-PRESERVE-NON-SECURE.N%d", node);
 	str = fdt_getprop(fdt, bdk_node, prop_name, &len);
-	if (str)
-		preserved_mem_size += (strtoul(str, NULL, 16) << 20);
+	if (str) {
+		preserved_mem_size = (strtoul(str, NULL, 16) << 20);
+		total_preserved_mem_size += preserved_mem_size;
+		if (preserved_mem_size)
+			printf("Preserved non-secure memory: 0x%llx bytes\n",
+			       preserved_mem_size);
+	}
 
 	snprintf(prop_name, 34, "DDR-CONFIG-PRESERVE-SECURE.N%d", node);
 	str = fdt_getprop(fdt, bdk_node, prop_name, &len);
-	if (str)
-		preserved_mem_size += (strtoul(str, NULL, 16) << 20);
+	if (str) {
+		preserved_mem_size = (strtoul(str, NULL, 16) << 20);
+		total_preserved_mem_size += preserved_mem_size;
+		if (preserved_mem_size)
+			printf("Preserved secure memory: 0x%llx bytes\n",
+			       preserved_mem_size);
+	}
 
-	if (preserved_mem_size)
-		printf("Total memory preserved region: 0x%llx bytes\n", preserved_mem_size);
-	return preserved_mem_size;
+	return total_preserved_mem_size;
 }
 
 const char *fdt_get_board_serial(void)
-- 
2.29.0

